src/components/pages/settings-page/edit-alert-dialog.tsx

```diff
@@ -0,0 +1,217 @@
+        let message = 'Internal Server Error';
+        if (error.response?.data?.errors && error.response.data.errors.length > 0) {
+          message = error.response.data.errors[0].message;
+        }
+        onError(message);
+      }
+      setFormDisabled(false);
+
+    }
+  };
+
+  const buildSettingBlock = () => {
+    if (!canSave()) {
+      return (
+        <Paper elevation={0} className={classes.invalidSettingBlock}>
+          <div>Invalid alert settings.</div>
+        </Paper>
+      );
+    }
+    const text = buildSettingText(notification!.settingsTemplate, {
+      id: setting!.id,
+      notificationId: setting!.notificationId,
+      threshold,
+      minTimeBetweenAlerts: minTimeBetweenAlerts * 60,
+      maxDailyCount,
+      smsEnabled,
+      emailEnabled,
+    });
+    return (
+      <Paper elevation={0} className={classes.settingBlock}>
+        {/* eslint-disable-next-line react/no-danger */}
+        <div dangerouslySetInnerHTML={{ __html: text }} />
+      </Paper>
+    );
+  };
+
+  return (
+    <Dialog className={classes.root} maxWidth="xs" onClose={() => onClose?.()} open={open}>
+      <DialogTitle id="alert-dialog-title">{notification!.name}</DialogTitle>
+      <DialogContent>
+        <Grid container spacing={3}>
+          <Grid item xs={6} className={classes.gridLabel}>
+            <Typography className={classes.label}>Alert me by:</Typography>
+          </Grid>
+          <Grid item xs={3}>
+            <FormControlLabel
+              control={(
+                <Checkbox
+                  checked={smsEnabled}
+                  disabled={formDisabled}
+                  onChange={(_, checked: boolean) => setSmsEnabled(checked)}
+                  name="SMS Enabled"
+                  color="primary"
+                />
+              )}
+              label="SMS"
+            />
+          </Grid>
+          <Grid item xs={3}>
+            <FormControlLabel
+              control={(
+                <Checkbox
+                  checked={emailEnabled}
+                  disabled={formDisabled}
+                  onChange={(_, checked: boolean) => setEmailEnabled(checked)}
+                  name="Email Enabled"
+                  color="primary"
+                />
+              )}
+              label="Email"
+            />
+          </Grid>
+          <Grid item xs={6} className={classes.gridLabel}>
+            <Typography className={classes.label}>Threshold:</Typography>
+          </Grid>
+          <Grid item xs={6}>
+            <TextField
+              className={classes.textField}
+              disabled={formDisabled}
+              required
+              onChange={(event: React.ChangeEvent<HTMLInputElement>) => setThreshold(Math.max(1, parseInt(event.target.value)))}
+              value={threshold}
+              type="number"
+              inputProps={{ min: '1', step: '1' }}
+            />
+          </Grid>
+          { notification!.defaultMaxDailyCount >= 0 && (
```

Comment:
Total nitpicking, but you can remove the extra space before each `notification` chunk.